005_hackerrank_challenges/001_introduction/004_list_replication/solution.hs

```diff
@@ -1,5 +1,10 @@
+#! /usr/bin/env nix-shell
+#! nix-shell ../../../004_an_updated_nix_shell_file/shell.nix  -i runghc
+
+-- run it locally by doing `cat sample-input.txt | ./solution.hs`
 
 f :: Int -> [Int] -> [Int]
+-- you could use `concatMap f x` instead of `concat (map f x)` here
 f n arr = concat $ map (replicate n) arr
```

Comment:
Suggestion given by hlint, ignoring the eta reduce hints since it clearer in the beginning